commons/zenoh-protocol-core/src/whatami.rs

```diff
@@ -259,12 +259,12 @@ impl BitOr for WhatAmIMatcher {
 impl BitOr for WhatAmI {
     type Output = WhatAmIMatcher;
     fn bitor(self, rhs: Self) -> Self::Output {
-        WhatAmIMatcher(unsafe { NonZeroU8::new_unchecked(self as u8 | rhs as u8) })
+        WhatAmIMatcher(unsafe { NonZeroU8::new_unchecked((self as u8 | rhs as u8) + 128) })
     }
 }
 
 impl From<WhatAmI> for WhatAmIMatcher {
     fn from(w: WhatAmI) -> Self {
-        WhatAmIMatcher(unsafe { NonZeroU8::new_unchecked(w as u8) })
+        WhatAmIMatcher(unsafe { NonZeroU8::new_unchecked(w as u8 + 128) })
```

Comment:
Same, use `|` instead of `+`

Reply:
To make things more explicit that we are actually setting the bit at position 7, the following could be used:
```rust
w as u8 | (1 << 7)
```